### DISCUSSION TITLE
(vim mode) is it possible to auto-save when you go from insert mode back to normal mode?

### DISCUSSION THREAD

**Discussion by IroncladDev** [zed-industries/zed#8121] [Feature Requests]
Auto-saving only when focus changes, when window focus changes, or on a repeating interval is a little weird for me.

I was wondering if there way a way to autosave only when you go from vim insert mode to normal mode.

**Comment by IroncladDev** [comment]
Update: I was able to get it to work by adding a binding to the `escape` key while in insert mode.

```
{
    "context": "Editor && vim_mode == insert && !menu",
    "bindings": {
      "escape": "workspace::Save"
    }
  }
```

**Reply by bosacky** [reply]
this was not working for me as it saves the file but does not exit the insert mode, so I came up with this approach which seems to be working:
```
  {
    "context": "Editor && vim_mode == insert && !menu",
    "bindings": {
      "escape": ["workspace::SendKeystrokes", "ctrl-c ctrl-s"]
    }
  }
```
